charmcraft/commands/pack.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import logging
+import os
+import pathlib
+import zipfile
+
+from charmcraft.cmdbase import BaseCommand, CommandError
+from .utils import load_yaml
+
+logger = logging.getLogger(__name__)
+
+# the minimum set of files in a bundle
+MANDATORY_FILES = {'bundle.yaml'}
+
+
+def build_zip(zippath, basedir, fpaths):
+    """Build the final file.
+
+    Note we convert all paths to str to support Python 3.5.
+    """
+    zipfh = zipfile.ZipFile(str(zippath), 'w', zipfile.ZIP_DEFLATED)
+    for fpath in fpaths:
+        zipfh.write(str(fpath), str(fpath.relative_to(basedir)))
+    zipfh.close()
+
+
+def get_paths_to_include(dirpath):
+    """Get all file/dir paths to include."""
+    allpaths = set()
+
+    # all mandatory files, which must exist (currently only bundles.yaml is mandatory, and
+    # it's verified before)
+    for fname in MANDATORY_FILES:
+        allpaths.add(dirpath / fname)
+
+    # the extra files, which must be relative
+    config = load_yaml(dirpath / 'charmcraft.yaml') or {}
+    prime_specs = config.get('parts', {}).get('bundle', {}).get('prime', [])
+
+    for spec in prime_specs:
+        if spec[0] == os.path.sep:
+            raise CommandError(
+                "Extra files in prime config can not be absolute: {!r}".format(spec))
+
+        fpaths = sorted(fpath for fpath in dirpath.glob(spec) if fpath.is_file())
+        logger.debug("Including per prime config %r: %s.", spec, fpaths)
+        allpaths.update(fpaths)
+
+    return sorted(allpaths)
+
+
+_overview = """
+Build the bundle, leaving a .zip file as the result of the process.
```

Comment:
Maybe something like:
```suggestion
Build the bundle and package it as a zip archive.
```

Reply:
Oh, very nice! Improved, thanks!